src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -144,30 +156,19 @@ Status ExperimentalMutableObjectManager::ReadRelease(const ObjectID &object_id)
   }
 
   auto &channel = reader_channel_entry->second;
-  RAY_RETURN_NOT_OK(EnsureGetAcquired(channel));
+  if (!channel.read_acquired) {
+    return Status::Invalid(
+        "Caller must call begin_read() before end_read() on a channel");
+  }
 
   RAY_RETURN_NOT_OK(
       channel.mutable_object->header->ReadRelease(channel.next_version_to_read));
   // The next read needs to read at least this version.
   channel.next_version_to_read++;
   channel.read_acquired = false;
 
-#endif
-  return Status::OK();
-}
+  sem_post(&channel.reader_semaphore);
```

Comment:
This means if release is not called, it can hang indefinitely? Is it okay in case of exceptions being raised? E.g., read 2 values, release 1 value and exception is raised

Reply:
Yes, but that was already true before. Exceptions get serialized as normal values.